markdown.sile-dev-1.rockspec

```diff
@@ -15,6 +15,7 @@ description = {
 }
 dependencies = {
    "lua >= 5.1",
+   "penlight >= 1.13.0", -- use of kpairs requires newer Penlight than transitive dependency possibly provided by SILE
```

Comment:
As noted in #78, the `markdown.sile` module does not use kpairs directly in its code, so this shouldn't be needed here.

As an aside note, modules preinstalled with SILE (penlight, lpeg, etc.), as of 0.14.x, do no show in `luarocks list`. It implies that any additional dependency added here will always be installed regardless of what SILE provides. I am not sure this is the way to go (if so, I'd need to add lpeg, but version ranges are problematic then, as it would mean other parts of SILE would have to be compatible with whatever version the module overloaded...)

Reply:
N.B. Not taken in #102

Reply:
> As an aside note, modules preinstalled with SILE (penlight, lpeg, etc.), as of 0.14.x, do no show in `luarocks list`. It implies that any additional dependency added here will always be installed regardless of what SILE provides.

This is only partially true. It will be true if SILE was configured and built `--without-system-luarocks` in which case it bundled vendored copies of them in it's own path. Otherwise `--with-system-luarocks` should interact nicely with `luarocks` including `luarocks list`. Just remember you do have to specify the Lua interpreter version to list for if it is not the default one LuaRocks is configured to use.

Reply:
We should probably add some sort of SILE subcommand for running LuaRocks with a matching version. https://github.com/sile-typesetter/sile/issues/1947

Reply:
Interesting, thanks for the explanation.
For the record, the route I took here was:
```
FROM docker.io/library/archlinux:latest
RUN pacman --needed --noconfirm -Syu
RUN pacman --noconfirm -S luarocks git
RUN pacman --noconfirm -Suy sile # it installed SILE v0.14.14 (LuaJIT 2.1.1702233742)
RUN luarocks list --lua-version=5.1
```

Typically, lpeg and penlight did not show as installed.
It might not be obvious for people installing pre-packaged versions of SILE as here (i.e. not building it themselves), how it was configured and built.

Usually I build my own version of SILE, so I'll try the above-mentioned option at some point.
(Here I just wanted a quick'n fresh ready-to-go docker image for testing)

Reply:
So the Dockerfile you show in that comment *will* show LPEG and Penlight because the Arch LInux system package for `sile` does use `--with-system-luarocks`. For convenience of people not knowing how to install LuaRocks to their system or not having a matching VM or whatever else, the default `./configure` option is actually `--without-system-luarocks`. That default is also what the 'official' SILE Docker image uses since it is a source build and does *not* use the Arch Linux packaged Lua libraries. It uses the exact versions specified in the rockspec installed in the vendored location.

We probably do need to rethink how that interacts with 3rd party modules like yours. As an example, if you used LPEG in your extension directly it would seem to make sense to add it to your rockspec, but that would actually make it harder to install or result in a duplicate installation where one of them is unnecessary. SILE will always provide LPEG whether on the system where LuaRocks also know about it or locally vendored where your module will be able to use it anyway. That isn't very evident or documented however.

Reply:
Note to self: rebuilding wholly the docker image clean (i.e. real latest stuff, no previous cached artifacts for earlier docker builds) = it worked indeed. Prolly I had some stuff cached.

Reply:
Also note since I've been reminding of this myself recently: the Dockerfile copies the entire source tree from your current Git project into the builder container. This can bite you if you have something configured and/or built or the host system. Ideally building Docker images would involve a pretty aggressive `git clean -dxff` before configuring just with enabling developer but not actually building anything on the host system before `make docker`.

I've wondered after this bit me a couple times if the Dockerfile should be forcing this cleanup after it copies the sources...